psh/echo/echo.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Phoenix-RTOS
+ *
+ * echo - display a line of text
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+void psh_echoinfo(void)
+{
+	printf("display a line of text");
+}
+
+
+static void psh_echo_help(const char *prog)
+{
+	printf("Usage: %s [options] [string]\n", prog);
+	printf("  -h:  shows this help message\n");
+}
+
+
+size_t psh_echoPrintVar(const char *var)
+{
+	size_t i = 1;
+
+	/* Right now only $? is supported */
+	if (*var != '?') {
+		while (*var != ' ' && *var != '\0') {
+			++var;
+			++i;
+		}
+	}
+	else {
+		/* ? is a special case - we eat only ?, even if no space is present */
+		printf("%d", psh_common.exitStatus);
+	}
+
+	return i; /* Eaten variable length */
+}
+
+
+int psh_echo(int argc, char **argv)
+{
+	int c, i;
+	size_t j;
+
+	while ((c = getopt(argc, argv, "h")) != -1) {
+		switch (c) {
+			case 'h':
+				psh_echo_help(argv[0]);
+				return EXIT_SUCCESS;
+
+			default:
+				psh_echo_help(argv[0]);
+				return EXIT_FAILURE;
+		}
+	}
+
+	for (i = optind; i < argc; ++i) {
+		if (i != 1)
```

Comment:
`if (i != optind)`. Probably won't matter until we add `-n` or `-e`

Reply:
Right, missed it, thx